core/core_bind.cpp

```diff
@@ -1900,4 +1900,61 @@ void EngineDebugger::_bind_methods() {
 	ClassDB::bind_method(D_METHOD("send_message", "message", "data"), &EngineDebugger::send_message);
 }
 
+////// LogManager //////
+// This is the client-facing part of the user log hooking system.
+// It's basically just a shunt that passes callbacks to the persistent UserLogManagerLogger singleton.
+
+LogManager *LogManager::singleton = nullptr;
+
+LogManager::LogManager() {
+	ERR_FAIL_COND_MSG(singleton != nullptr, "Somehow created two LogManagers");
+
+	singleton = this;
+}
+
+LogManager::~LogManager() {
+	ERR_FAIL_COND_MSG(singleton != this, "LogManager::singleton not correct on exit");
+
+	singleton = nullptr;
+}
+
+void LogManager::register_log_capture_nonthreadsafe(const Callable &p_callable) {
+	UserLogManagerLogger *log_manager = UserLogManagerLogger::get_singleton();
+	ERR_FAIL_COND_MSG(log_manager == nullptr, "log_manager not yet initialized; this shouldn't be possible");
+	if (log_manager != nullptr) {
+		log_manager->register_log_capture_nonthreadsafe(p_callable);
+	}
+}
+
+void LogManager::unregister_log_capture_nonthreadsafe(const Callable &p_callable) {
+	UserLogManagerLogger *log_manager = UserLogManagerLogger::get_singleton();
+	ERR_FAIL_COND_MSG(log_manager == nullptr, "log_manager not yet initialized; this shouldn't be possible");
+	if (log_manager != nullptr) {
+		log_manager->unregister_log_capture_nonthreadsafe(p_callable);
+	}
+}
+
+void LogManager::register_log_capture_buffered(const Callable &p_callable) {
+	UserLogManagerLogger *log_manager = UserLogManagerLogger::get_singleton();
+	ERR_FAIL_COND_MSG(log_manager == nullptr, "log_manager not yet initialized; this shouldn't be possible");
+	if (log_manager != nullptr) {
+		log_manager->register_log_capture_buffered(p_callable);
+	}
+}
+
+void LogManager::unregister_log_capture_buffered(const Callable &p_callable) {
+	UserLogManagerLogger *log_manager = UserLogManagerLogger::get_singleton();
+	ERR_FAIL_COND_MSG(log_manager == nullptr, "log_manager not yet initialized; this shouldn't be possible");
+	if (log_manager != nullptr) {
+		log_manager->unregister_log_capture_buffered(p_callable);
+	}
+}
+
+void LogManager::_bind_methods() {
+	ClassDB::bind_method(D_METHOD("register_log_capture_nonthreadsafe", "callable"), &LogManager::register_log_capture_nonthreadsafe);
+	ClassDB::bind_method(D_METHOD("unregister_log_capture_nonthreadsafe", "callable"), &LogManager::unregister_log_capture_nonthreadsafe);
+	ClassDB::bind_method(D_METHOD("register_log_capture_buffered", "callable"), &LogManager::register_log_capture_buffered);
+	ClassDB::bind_method(D_METHOD("unregister_log_capture_buffered", "callable"), &LogManager::unregister_log_capture_buffered);
```

Comment:
The name `nonthreadsafe` looks out of place to me. I suggest `non_thread_safe` as we already use `thread_safe` in many other places throughout the engine.

Reply:
Done!